### ISSUE TITLE
trace::SpanId doesn't support storing in maps

### ISSUE THREAD

**Issue by VivekSubr** [open-telemetry/opentelemetry-cpp#1888]
**Is your feature request related to a problem?**
trace::SpanId lacks comparision operators so can't be used with std::map, and no hash specialization so can't be used with std::unordered_map

**Describe the solution you'd like**
Addition of comparison operator and hash specialization to SpanId class.

**Describe alternatives you've considered**
The problem is - how to suspend spans for multithreaded applications, ie, a span has to be paused and resumed... one way is to store in a global map, but I don't see any other answer/api in opentelemetry itself?

**Additional context**
Basically, need a way to 'pause' and 'resume' a span.

**Comment by lalitb** [comment]
@VivekSubr 
 - Can you provide a use-case for pausing and resuming a span, and need to store them in std::map?
 - Trace API is Stable, and any new change would be ABI breaking, so would be difficult to accommodate in general.

**Comment by VivekSubr** [comment]
@lalitb 
Yeah, it's like say you have a class wrapping opentelemetry apis, and the app needs to yield it's span (pause) to other threads/operations, and restart it latter.So, at that point, we'll store it in a map for latter retrieval. 

It's just addition of operators, not a entire new api as such

**Comment by lalitb** [comment]
@VivekSubr - Can you use the custom comparator for TraceId for storing in std::map? In general, we would like to avoid new API changes (even if it is addition of new method/operator), if there are alternate solution.

**Comment by VivekSubr** [comment]
@lalitb - yeah, I guess that works... but this makes library quite opinionated right, with how it's supposed to be used? I can raise a PR adding the operators sometime if you want.

**Comment by lalitb** [comment]
@VivekSubr  - It's actually more than the implementation effort. Addition new methods can (sometime) break the ABI compatibility between otel-api and otel-sdk. We want to ensure that existing instrumentation libraries (compiled against the older API) are linking successfully with the application (compiled against the newer SDK). And vice versa. Feel free to raise a PR, and this can be further discussed there with other reviewers/approvers.

**Comment by VivekSubr** [comment]
Okay, closing for now
